### ISSUE TITLE
[Popup] Popup Should Be Next to settings.target if used

### ISSUE THREAD

**Issue by Parakleta** [Semantic-Org/Semantic-UI#3093]
If a Popup is marked as inline with a specific target then it is expected that the `.ui.popup` selector will find the inline elements as a later sibling of the target (this is fine).  Unfortunately after that it stores the root popup element in `settings.popup` which disables much of the inline popup specific code.

Additionally to this, when a non-inline popup is inserted into the DOM it is appended to the parent of the activating element instead of the parent of the target.  These interact to take inline popup elements from being a sibling of the target to being a sibling of the trigger.

See [popup.js](https://github.com/Semantic-Org/Semantic-UI/blob/6f4ac8697b3488afe4582a7129cca61bb0eccc2c/src/definitions/modules/popup.js#L98-L116).

**Comment by jlukic** [comment]
There are many edge cases which this particular set of constraints is used for (which have accumulated from having this component around in the lib for the last few years).

If there's a specific issue with placement/layout you can help me out by creating a test case.

I recommend in general specifying an existing popup with `settings.popup` if possible, the auto-discovery feature was added to make the code simpler in many cases, but its not as granular.

**Comment by Parakleta** [comment]
I'll have to wait till Monday to set up a test case for you, but the general suggestion I would make is that I think the pop up should be appended to the target's parent and not the trigger's.  The case I have is a popup triggered by the `.link.icon` of an `.icon.input`, targeting the `.icon.input`.  The problem is the popup is being pulled inside the `.icon.input` instead of being beside it and this is messing up the formatting of other inputs within the popup.  This problem is really irrespective of the use of `inline` but there is code to protect against this case if `inline` is true, but the code is disabled by the fact that the inline popup is stored in `settings.popup`.

**Comment by Parakleta** [comment]
Here's the [fiddle](http://jsfiddle.net/parakleta/j15qk2pz/2/).

Observe how if you click the clock icon to bring up the popup when you go to type into the text field in the popup your cursor is offset by the icon size even though the input on the popup sheet is not an `.icon.input`.  This is because the popup has been dragged into the `.icon.input` to be a sibling of the `.link.icon` so all of its inputs become `.icon.input` as well.

I think the popup should always be a sibling of the target not the trigger.

**Comment by Parakleta** [comment]
For my case (not sure if there are side effects) simply removing the [assignment to `settings.popup`](https://github.com/Semantic-Org/Semantic-UI/blob/c579a4e7a6d81c43af87ab198dc3b2788d0a1252/src/definitions/modules/popup.js#L100) for inline popups fixed it.

**Comment by jlukic** [comment]
It's clear to me that popup should be positioned next to `settings.target` (because the target might have a different coordinate system than activator), so that is indeed a bug. 

I'm unsure of how `settings.preserve` play into what you're describing. I'm also unsure what bug I'm suppose to be observing in the test case. Can you help clarify?

**Comment by Parakleta** [comment]
The fiddle was just an example of my use case and how I found the problem.  The popup is pulled inside the `.icon.input` and this changes the styling of the input in the popup.  It's not the best example but it was what I had to hand.

The change of `settings.popup = $popup` to `settings.preserve = true` was just a work-around.  It was the simplest change to make the code work for my case, as it triggered inline popup handling code which does the correct thing re: target but is excluded if `settings.popup` is set.

I think the intended behaviour of inline popups being stored in `settings.popup` is correct in general, and then `preserve` has orthogonal behaviour, so you can ignore those suggestions.  The `refresh` function really needs refactoring but once that's done I think the rest should be fine.

**Comment by jlukic** [comment]
I'm unable to determine a specific required change for how popup is handled. I'd be happy to work at fixing specific test cases where popup is pulling the wrong element or offset.

Feel free to open another issue if that is the case.

**Comment by Parakleta** [comment]
I don't understand your closure message.  I have show there is dead code that needs to be restored for correct behaviour, I provided a specific required change that resolves the issue for me (although I am unaware of possible side effects of this change), and I have provided a test case which shows the popup is being moved (and misplaced) in the DOM which causes CSS problems.

What else do I need to provide?
